### PR TITLE
tailcfg: import ProtoPortRange for local use

### PR DESCRIPTION
**Pull request by raggi**
Imported type and parsing, with minor modifications.

Updates https://github.com/tailscale/corp/issues/15043

### CODE REVIEW

**Review by twitchyliquid64 — APPROVED** [review]

File: tailcfg/proto_port_range.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"tailscale.com/types/ipproto"
+	"tailscale.com/util/vizerror"
+)
+
+// ProtoPortRange is used to encode "proto:port" format in a GrantRow.IP field.
```
Comment (maisem) [inline tailcfg/proto_port_range.go:16]: remove reference to `GrantRow`

File: tailcfg/proto_port_range.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+		return "", nil, vizerror.New("host cannot contain a comma (\",\")")
+	}
+
+	if portlist == "*" {
+		// Special case: permit hostname:* as a port wildcard.
+		ports = append(ports, PortRangeAny)
+		return host, ports, nil
+	}
+
+	pl := strings.Split(portlist, ",")
```
Comment (maisem) [inline tailcfg/proto_port_range.go:148]: This was only relevant for parsing comma separated lists of ports like `80-90,443-450,600`. I don't think we need that here

File: types/ipproto/protoname.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	"igp":       9,
+	"ip-in-ip":  4, // IANA says "ipv4"; Wikipedia/popular use says "ip-in-ip"
+	"ipv4":      4,
+	"ipv6-icmp": ICMPv6,
+	"sctp":      SCTP,
+	"tcp":       TCP,
+	"udp":       UDP,
+}
+
+func ResolveProtoName(s string) (p Proto, ok bool, err error) {
```
Comment (maisem) [inline types/ipproto/protoname.go:33]: docs

File: tailcfg/proto_port_range.go (hunk 2 above)
Comment (raggi) [reply tailcfg/proto_port_range.go:148]: I would like to not duplicate this code, and this is a requirement of other locations in which it is used.

File: tailcfg/proto_port_range.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+	"fmt"
+	"strconv"
+	"strings"
+
+	"tailscale.com/types/ipproto"
+	"tailscale.com/util/vizerror"
+)
+
+// ProtoPortRange is used to encode "proto:port" format in an ACL IP field.
+// The following formats are supported:
```
Comment (maisem) [inline tailcfg/proto_port_range.go:17]: I meant lets not mention ACLs at all.

**Review by maisem — APPROVED** [review]

**Pull request merged**